src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+            }
+        }
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        // create quorum with the first randomSplit elements
+        cheats.startPrank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers1);
+
+        // add the rest of the strategies
+        for (uint i = 0; i < strategiesAndWeightingMultipliers2.length; i++) {
+            cheats.expectEmit(true, true, true, true, address(voteWeigher));
+            emit StrategyAddedToQuorum(quorumNumber, strategiesAndWeightingMultipliers2[i].strategy, strategiesAndWeightingMultipliers2[i].multiplier);
+        }
+        voteWeigher.addStrategiesConsideredAndMultipliers(quorumNumber, strategiesAndWeightingMultipliers2);
+
+        // check that the quorum was created and strategies were added correctly
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            IVoteWeigher.StrategyAndWeightingMultiplier memory strategyAndWeightingMultiplier = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumNumber, i);
+            assertEq(address(strategyAndWeightingMultiplier.strategy), address(strategiesAndWeightingMultipliers[i].strategy));
+            assertEq(strategyAndWeightingMultiplier.multiplier, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+    }
+
+    function testAddStrategiesConsideredAndMultipliers_NotFromServiceManagerOwner_Reverts(
+        address notServiceManagerOwner,
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        cheats.assume(notServiceManagerOwner != serviceManagerOwner);
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+        // we need 2 or more strategies to create the quorums and then add one
+        cheats.assume(strategiesAndWeightingMultipliers.length > 1);
+
+        // separate strategiesAndWeightingMultipliers into 2 arrays, 1 with the last element and 1 with the rest
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers1 = new IVoteWeigher.StrategyAndWeightingMultiplier[](strategiesAndWeightingMultipliers.length - 1);
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers2 = new IVoteWeigher.StrategyAndWeightingMultiplier[](1);
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length - 1; i++) {
+            strategiesAndWeightingMultipliers1[i] = strategiesAndWeightingMultipliers[i];
+        }
+        strategiesAndWeightingMultipliers2[0] = strategiesAndWeightingMultipliers[strategiesAndWeightingMultipliers.length - 1];
+
+        // create quorum with all but the last element
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.prank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers1);
+
```

Comment:
I _suspect_ that you don't actually need all of this setup, based on the order in which reverts will occur. That being said, this seems good since it's less brittle, because changing the order of reverts shouldn't break it.

Reply:
fixed